presto-main/src/main/java/com/facebook/presto/metadata/PartitioningMetadata.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.metadata;
+
+import com.facebook.presto.sql.planner.PartitioningHandle;
+import com.google.common.collect.ImmutableList;
+
+import java.util.List;
+
+import static java.util.Objects.requireNonNull;
+
+public class PartitioningMetadata
```

Comment:
hmm.. the motivation to have to `PartitioningMetadata` and `ConnectorPartitioningMetadata` is so one is only visible  to engine, one is only visible to SPI?

Reply:
`ConnectorPartitioningMetadata` contains `ConnectorPartitioningHandle`, `PartitioningMetadata` has `PartitioningHandle`.

Reply:
So one is for engine, one is for SPI ? (like other X vs. ConnectorX)